src/portable/portablectl.c

```diff
@@ -439,6 +526,35 @@ static int attach_image(int argc, char *argv[], void *userdata) {
         (void) maybe_reload(&bus);
 
         print_changes(reply);
+
+        if (arg_enable || arg_now) {
+                r = sd_bus_message_rewind(reply, 1);
+                if (r < 0)
+                        return r;
+                r = sd_bus_message_enter_container(reply, 'a', "(sss)");
+                if (r < 0)
+                        return bus_log_parse_error(r);
+
+                for (;;) {
+                        char *type, *path, *source;
+
+                        r = sd_bus_message_read(reply, "(sss)", &type, &path, &source);
+                        if (r < 0)
+                                return bus_log_parse_error(r);
+                        if (r == 0)
+                                break;
+
+                        if ((streq(type, "symlink") || streq(type, "copy")) && endswith(path, ".service")) {
```

Comment:
why only the .service unit type?

There's STR_IN_SET() to make the repeated string comparisons prettier

Reply:
Because I have only used portable images with services :-) Are other types of unit supported too?

Reply:
yes. not mount units and stuff, but sockets and targets definitely.

Reply:
Thanks, will add sockets and targets.